google-cloud-debugger/docs/instrumentation.md

```diff
@@ -0,0 +1,136 @@
+    Doc](https://cloud.google.com/debugger/docs/debugging) on how to
+    manage breakpoints on Cloud Console.
+
+When this library is installed and configured in your running application,
+you can view your applications breakpoint snapshots in real time by
+opening the Google Cloud Console in your web browser and navigating to the
+"Debug" section. It also integrates with Google App Engine Flexible to
+make application registration more seemless, and helps Stackdriver
+Debugger Console to select correct version of source code from Cloud
+Source Repository.
+
+Note that when no breakpoints are created, the debugger agent consumes
+very little resource and has no interference with the running application.
+Once breakpoints are created and depends on where the breakpoints are
+located, the debugger agent may add a little latency onto each request.
+The application performance will be back to normal after all breakpoints
+are finished being evaluated. Be aware the more breakpoints are created,
+or the harder to reach the breakpoints, the more resource the debugger
+agent would need to consume.
+
+### Using instrumentation with Ruby on Rails
+
+To install application instrumentation in your Ruby on Rails app, add this
+gem, `google-cloud-debugger`, to your Gemfile and update your bundle. Then
+add the following line to your `config/application.rb` file:
+
+```ruby
+require "google/cloud/debugger/rails"
+```
+
+This will load a Railtie that automatically integrates with the Rails
+framework by injecting a Rack middleware. The Railtie also takes in the
+following Rails configuration as parameter of the debugger agent
+initialization:
+
+```ruby
+# Explicitly enable or disable Stackdriver Debugger Agent
+config.google_cloud.use_debugger = true
+# Shared Google Cloud Platform project identifier
+config.google_cloud.project_id = "gcloud-project"
+# Google Cloud Platform project identifier for Stackdriver Debugger only
+config.google_cloud.debugger.project_id = "debugger-project"
+# Shared Google Cloud authentication json file
+config.google_cloud.keyfile = "/path/to/keyfile.json"
+# Google Cloud authentication json file for Stackdriver Debugger only
+config.google_cloud.debugger.keyfile = "/path/to/debugger/keyfile.json"
+# Stackdriver Debugger Agent module name identifier
+config.google_cloud.debugger.module_name = "my-ruby-app"
+# Stackdriver Debugger Agent module version identifier
+config.google_cloud.debugger.module_version = "v1"
+```
+
+See the Google::Cloud::Debugger::Railtie class for more information.
+
+### Using instrumentation with Sinatra
+
+To install application instrumentation in your Sinatra app, add this gem,
+`google-cloud-debugger`, to your Gemfile and update your bundle. Then add
+the following lines to your main application Ruby file:
+
+```ruby
+require "google/cloud/debugger"
+use Google::Cloud::Debugger::Middleware
+```
+
+This will install the debugger middleware in your application.
```

Comment:
I'll add something to show how to pass configuration as parameters. But some of it may change down the road.